modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -274,49 +294,50 @@ MultiSmoothSuperellipsoidIC::overlapCheck(const Point & newcenter, Real nc_as, R
-          else if (pc == 2)
-            tmp_p(1) -= _bs[el];
-          else if (pc == 3 )
-            tmp_p(1) += _bs[el];
-          else if (pc == 4 )
-            tmp_p(2) -= _cs[el];
-          else
-            tmp_p(2) += _cs[el];
-
-          Real dist = _mesh.minPeriodicDistance(_var.number(), tmp_p, l_p);
-          //When dist is 0 we are exactly at the center of the superellipsoid so return _invalue
-          //Handle this case independently because we cannot calculate polar angles at this point
-          if (dist == 0.0)
-            return 0.0;
-
-          //Compute the distance r from the center of the superellipsoid to its outside edge
-          //along the vector from the center to the current point
-          //This uses the equation for a superellipse in polar coordinates and substitutes
-          //distances for sin, cos functions
-          Point dist_vec = _mesh.minPeriodicVector(_var.number(), l_p, tmp_p);
-
-          //First calculate rmn = r^(-n), replacing sin, cos functions with distances
-          Real rmn = (std::pow(std::abs(dist_vec(0) / dist / a), n)
-                    + std::pow(std::abs(dist_vec(1) / dist / b), n)
-                    + std::pow(std::abs(dist_vec(2) / dist / c), n) );
-          //Then calculate r2 from rmn
-          Real r2 = std::pow(rmn, (-1.0/n));
-
-          if (dist < r2)
-            return 0.0;
-        }
+        tmp_p = _centers[el];
+        //Find extremes along semiaxis of accepted ellipsoids
+        if (pc == 0)
+          tmp_p(0) -= _as[el];
+        else if (pc == 1)
+          tmp_p(0) += _as[el];
+        else if (pc == 2)
+          tmp_p(1) -= _bs[el];
+        else if (pc == 3 )
+          tmp_p(1) += _bs[el];
+        else if (pc == 4 )
+          tmp_p(2) -= _cs[el];
+        else
+          tmp_p(2) += _cs[el];
+
+        Real dist = _mesh.minPeriodicDistance(_var.number(), tmp_p, l_p);
+        //When dist is 0 we are exactly at the center of the superellipsoid so return _invalue
+        //Handle this case independently because we cannot calculate polar angles at this point
+        if (dist == 0.0)
+          return 0.0;
+
+        //Compute the distance r from the center of the superellipsoid to its outside edge
+        //along the vector from the center to the current point
+        //This uses the equation for a superellipse in polar coordinates and substitutes
+        //distances for sin, cos functions
+        Point dist_vec = _mesh.minPeriodicVector(_var.number(), l_p, tmp_p);
+
+        //First calculate rmn = r^(-n), replacing sin, cos functions with distances
+        Real rmn = (std::pow(std::abs(dist_vec(0) / dist / a), n)
+                  + std::pow(std::abs(dist_vec(1) / dist / b), n)
+                  + std::pow(std::abs(dist_vec(2) / dist / c), n) );
+        //Then calculate r2 from rmn
+        Real r2 = std::pow(rmn, (-1.0/n));
+
+        if (dist < r2)
+          return 0.0;
       }
     }
+  }
   return 1.0;
```

Comment:
This is not a the appropriate return type. The function returns a "status" as a Real which is either 0.0 or 1.0. The correct return type should be `bool` (and if more than just overlaps yes/no needs to be encoded, it should be an enum).